src/main/java/Cache.java

```diff
@@ -0,0 +1,74 @@
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Cache {
```

Comment:
does it need to be public? class and methods within it

Reply:
In this case it can be package private (default). I make it public because of habit in my current work, we doesn't use default because it is less intuitive in project with a lot of modules and we need change it to public if we want use it outside of package in new task.